nexus/db-model/src/omicron_zone_config.rs

```diff
@@ -46,6 +51,8 @@ pub(crate) struct OmicronZone {
     pub(crate) snat_ip: Option<IpNetwork>,
     pub(crate) snat_first_port: Option<SqlU16>,
     pub(crate) snat_last_port: Option<SqlU16>,
+    // Only present for BlueprintZoneConfig; always `None` for OmicronZoneConfig
+    pub(crate) external_ip_id: Option<ExternalIpUuid>,
```

Comment:
It's a good question. Maybe flippant, but I think my response would be "this whole file is pretty suspect and probably deserves rethinking". It was something I threw together to handle the fact that blueprints and inventory both need to serialize/deserialize omicron zones into the DB, but their types aren't unified despite having identical contents. As they diverge over time this will probably get unreasonably messy; arguably it was overly optimistic to try to unify them in the first place?